### PR TITLE
Allow usage of `proof_assert` in closures

### LINKED ISSUE #1072: Capturing a variable in a closure introduces a non-local variable in `DebugVarInfo`
More specifically, Creusot assumes that `DebugVarInfo` only contains local variables, but then the following crashes:

```rust,creusot
fn crashes() {
    let mut x = Vec::new();
    let _closure = || {
        x.push(42);
        let prev = Some(3);
        proof_assert!(prev != None);
    };
}
```

This triggers an "called `Option::unwrap()` on a `None` value" error in `creusot::translation::specification::ScopeTree::build`, caused by the capture of `x`.

### PR DESCRIPTION
**Pull request by arnaudgolfouse**
Should fix #1072

### CODE REVIEW

**Comment by jhjourdan** [comment]
Could you add some tests?

File: creusot/tests/should_succeed/ghost/assert_in_ghost.coma [hunk 1]
```diff
@@ -0,0 +1,981 @@
...
+    [ return' (result:GhostBox'0.t_ghostbox ())-> {[%#span6] inv'1 result}
+      {[%#span5] CreusotContracts_Ghost_GhostBox_Type.ghostbox_0 result = x}
+      (! return' {result}) ]
+    
+  
+  use prelude.prelude.Int32
+  
+  use AssertInGhost_GhostCapture_Closure0_Type as Closure'0
+  
+  predicate resolve'1 [#"/home/arnaud/Documents/Programmation/Rust/CONTRIBUTIONS/creusot/creusot/tests/should_succeed/ghost/assert_in_ghost.rs" 14 4 17 5] (_1 : Closure'0.assertinghost_ghostcapture_closure0)
```
Comment (xldenis) [inline creusot/tests/should_succeed/ghost/assert_in_ghost.coma:327]: huh is it normal that we are storing absolute paths in tests @Armael ?

Comment (Armael) [reply creusot/tests/should_succeed/ghost/assert_in_ghost.coma:327]: no, that's not expected, of course

Comment (jhjourdan) [reply creusot/tests/should_succeed/ghost/assert_in_ghost.coma:327]: @arnaudgolfouse How have you generated these files? You should use `cargo test --test ui assert_in_ghost -- --bless`

Comment (arnaudgolfouse) [reply creusot/tests/should_succeed/ghost/assert_in_ghost.coma:327]: Ah yes, that's my fault, I forgot about this

**Comment by arnaudgolfouse** [comment]
> Could you add some tests?

I added some very basic tests, if you have a more complicated idea in mind don't hesitate

File: creusot/tests/should_succeed/closures/09_proof_assert_in_closure.rs [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+}
+
+pub fn mutable_capture() {
+    let mut x = 1;
+    let _clos = #[requires(x == 1i32)]
+    || {
+        proof_assert!(x == 1i32);
+        x = 2;
+        proof_assert!(x == 2i32);
+    };
```
Comment (jhjourdan) [inline creusot/tests/should_succeed/closures/09_proof_assert_in_closure.rs:19]:
```suggestion
    } ();
```

**Comment by jhjourdan** [comment]
I'm not asking anything complicated. Jsut a basic test to check that this feature still works in the future.

**Comment by jhjourdan** [comment]
This looks good. Let's merge.

**Comment by arnaudgolfouse** [comment]
Why is list_reversal_lasso failing? It seems to be unrelated, and it passes on my machine...

**Comment by jhjourdan** [comment]
Is it failing? If the PR has been mergend, this means that it has succeeded.

**Pull request merged**